builtin/providers/aws/resource_aws_route.go

```diff
@@ -0,0 +1,177 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+// AWS Route resource Schema delcaration
+func resourceAwsRoute() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsRouteCreate,
+		Read:   resourceAwsRouteRead,
+		Update: resourceAwsRouteUpdate,
+		Delete: resourceAwsRouteDelete,
+
+		Schema: map[string]*schema.Schema{
+			"destination_cidr_block": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+
+			"destination_prefix_list_id": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"gateway_id": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+
+			"instance_id": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+
+			"instance_owner_id": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"network_interface_id": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+
+			"origin": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"state": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"route_table_id": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+
+			"vpc_peering_connection_id": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+		},
+	}
+}
+
+func resourceAwsRouteCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).ec2conn
+
+	if d.Get("route_table_id") == "" {
+		fmt.Errorf("Error: no route table ID specified to add the route to." +
+			"Please specify the ID of the routing table to add the route to.")
+
+	}
+
+	// Create the route
+	createOpts := &ec2.CreateRouteInput{
+		DestinationCIDRBlock:   aws.String(d.Get("destination_cidr_block").(string)),
+		GatewayID:              aws.String(d.Get("gateway_id").(string)),
+		InstanceID:             aws.String(d.Get("instance_id").(string)),
+		NetworkInterfaceID:     aws.String(d.Get("network_interface_id").(string)),
+		RouteTableID:           aws.String(d.Get("route_table_id").(string)),
+		VPCPeeringConnectionID: aws.String(d.Get("vpc_peering_connection_id").(string)),
```

Comment:
They're not all required -- only the `destination_cidr_block` field is. The target type can vary. I was thinking check if more than one target type is set and if it is, error out? Because it can only be one destination